Retrieving Cloud Storage Buckets Data (List & Details)

kangwork opened the pull request:
**Description:**
- Completed `get_storage_buckets.py ` using fastAPI
    - This module provides APIs for 
        (1) getting the list of all buckets in a project and 
        (2) getting the details of a specific bucket.

- Test results and other logs will be documented on the project log asap tomorrow

**Other Updates:**
- Could not resolve the problems with the `get_iam_roles.py`'s imports but will do it by tomorrow and push it
- Did not modify `credentials.py` and `get_instance_details.py` file today

**TODOs:**
- Update  `credentials.py`
- Complete `get_iam_roles.py`
- Document formally on Confluence

kangwork committed:
Wrote some comments and a new function to get details of a bucket with a given name
Need to be tested and revised on a company laptop

kangwork committed:
Completed get_storage_buckets.py except the auto deployment part

kangwork committed:
Commited the part undone

ImMin5 commented on get_storage_buckets.py:
```diff
@@ -3,26 +3,87 @@
+# # 3. Run the app
 if __name__ == '__main__':
-    uvicorn.run(app, host="
+    print("Manual Deployment:")
+    print("The command `uvicorn get_storage_buckets:app --reload` will run the app")
+    print("You can visit http://localhost:8000 to check the app")
+    print("Press Ctrl+C to stop the app")
```
print 말고 python의 logging 패키지를 이용해서 로그를 기록해 주세요

kangwork replied:
넵 확인하였습니다!
전에 푸시 된 irene-iam 브랜치의 `get_iam_roles.py` 또한 마찬가지로 바꿔서 PR 드리겠습니다.

kangwork replied:
![image](https://github.com/kangwork/MiniGoogleCloudCollector/assets/128764986/641406d0-39b7-41aa-b881-f6cd19a5a65b)

이런 식으로 디스플레이 되면 괜찮을까요? 감사합니다!
--> 다시 바꿔보겠습니다!

ImMin5 commented on get_storage_buckets.py:
```diff
@@ -3,26 +3,87 @@
 
-# 2. Define the routes
+# Some other global variables
+project_id = "bluese-cloudone-20200113"
```
여러개의 프로젝트 id의 대한 credential을 사용할 수 있다는 가정하에 
`project_id`도 동적으로 가져올 수 있게 변경이 되었으면 좋을 것 같습니다.

kangwork replied:
넵 확인하였습니다!

kangwork replied:
현재는 url query parameter로 동적으로 가능하지만, 유저 측에서 제공되지 않았을 때 디폴트를 받는 형식으로 만들었습니다. 제가 변수명을 직관적이지 않게 만든 것 같아요.
혹시 다른 방식이 추천된다면 편히 알려주세요. 변경하겠습니다! 감사합니다.

kangwork committed:
Replaced the print statements to the use of a python's logging package,
Added some comments that explain the fact that the global variable 'default_project_id' serves as an example default, but the user can specify the id by giving a query parameter, and
Changed the variable name to be more intuitive (i.e. project_id --> default_project_id)
